components/layout/fragment.rs

```diff
@@ -655,10 +655,11 @@ pub struct ScannedTextFragmentInfo {
     pub content_size: LogicalSize<Au>,
 
     /// The position of the insertion point in characters, if any.
-    ///
-    /// TODO(pcwalton): Make this a range.
     pub insertion_point: Option<CharIndex>,
 
+    /// Is this fragment selected?
+    pub selected: bool,
+
```

Comment:
Can we move this down to after `requires_line_break_afterward_if_wrapping_on_newlines`? (Or, even better, combine those two into a `bitflags!`?)

Fragment memory usage tends to have an outsized effect on Servo's performance, since they do get moved around a lot (more than they should, but fixing this will be a good bit of work).